app/assets/javascripts/Components/submission_file_manager.jsx

```diff
@@ -1,14 +1,16 @@
 import React from 'react'
 import { render } from 'react-dom'
 import FileManager from './markus_file_manager'
+import SubmissionFileUploadModal from './Modals/submission_file_upload_modal'
 
 
 class SubmissionFileManager extends React.Component {
 
   constructor(props) {
     super(props);
     this.state = {
-      files: []
+      files: [],
+      showModal: false
```

Comment:
Seems like you need to add an `uploadTarget` state as well?

Reply:
it's not necessary but it's good to add so the reader is aware of all possible state keys.